pennylane/templates/subroutines/multiplier.py

```diff
@@ -269,25 +270,42 @@ def _multiplier_decomposition_resources(
     else:
         num_wires_aux = num_x_wires
 
-    resources = {
-        resource_rep(QFT, num_wires=num_wires_aux): 2,
-        resource_rep(
-            ControlledSequence,
-            base_class=PhaseAdder,
-            base_params={"num_x_wires": num_wires_aux, "mod": mod},
-            num_control_wires=num_x_wires,
-        ): 1,
-        adjoint_resource_rep(
-            ControlledSequence,
-            {
-                "base_class": PhaseAdder,
-                "base_params": {"num_x_wires": num_wires_aux, "mod": mod},
-                "num_control_wires": num_x_wires,
-            },
-        ): 1,
-        adjoint_resource_rep(QFT, {"num_wires": num_wires_aux}): 2,
-        SWAP: num_x_wires,
+    cs_base_params = {
+        "base_class": PhaseAdder,
+        "base_params": {"num_x_wires": num_wires_aux, "mod": mod},
+        "num_control_wires": num_x_wires,
+    }
+
+    params = {
+        "compute_op_params": {
+            "compute_op": resource_rep(QFT, num_wires=num_wires_aux),
+            "target_op": resource_rep(ControlledSequence, **cs_base_params),
+            "uncompute_op": resource_rep(
+                Adjoint, base_class=QFT, base_params={"num_wires": num_wires_aux}
+            ),
+        },
+        "target_op_params": {
+            "resources": {resource_rep(SWAP): num_x_wires},
+        },
+        "uncompute_op_params": {
+            "compute_op": resource_rep(QFT, num_wires=num_wires_aux),
+            "target_op": resource_rep(
+                Adjoint, base_class=ControlledSequence, base_params=cs_base_params
+            ),
+            "uncompute_op": resource_rep(
+                Adjoint, base_class=QFT, base_params={"num_wires": num_wires_aux}
+            ),
+        },
     }
+    if num_x_wires > 1:
```

Comment:
Could consider setting the `"target_op_params"` for this branch only within the `if` clause, to have a more symmetric setup and avoid overwriting data. Not important though.